### PR TITLE
Add Introduction to Pointers

### PR DESCRIPTION
**Pull request by jerrettl**
A starting point. This first portion is just the prelude to the introduction.

### CODE REVIEW

**Comment by jerrettl** [comment]
At last, the entire introduction is finished.

File: src/intro-to-c/pointers/intro-to-pointers.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Introduction to Pointers
+
+*Created by Jerrett Longworth in March 2022.*
+
+This is where things start to get really crazy. With the concept of pointers, you are given more power, but it requires great responsibility.
```
Comment (nizolnier) [inline src/intro-to-c/pointers/intro-to-pointers.md:5]: love the reference

File: src/intro-to-c/pointers/using-pointers.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+These are all valid, but is important as different coders prefer different styles. Any of these styles are purely for visual understanding and are treated identically once compiled and executed.
+
+---
+
+## Pass-by-Value vs. Pass-by-Reference
+
+This is where the power of pointers starts to come into play! By passing pointers into functions, they can directly use and modify the values pointed to, no longer requiring you to return something!
+
+When passing variables into functions, there are two categories of passing: passing by value and passing by reference. The main difference is simply whether or not you are passing a pointer into a function.
```
Comment (nizolnier) [inline src/intro-to-c/pointers/using-pointers.md:155]: :o that makes a lot of sense lol

File: src/intro-to-c/pointers/using-pointers.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+Using the reference and dereference operators is cool, but there is one more piece of the puzzle missing. How can you *store* an address and use it later?
+
+For all of the data types seen before, like `int`, `double`, `char`, and so on, placing an asterisk (`*`) after them indicates a pointer to that respective data type. For example, `int*` represents a pointer to an `int`, and `double*` represents a pointer to a `double`.
+
+*Note: Yes, this can be confusing at first. The same symbol is used for both declaring pointers and the operation for dereferencing pointers.*
+
+With that aside, let's see a trivial example of how a pointer is created:
+
+``` c
```
Comment (nizolnier) [inline src/intro-to-c/pointers/using-pointers.md:69]: if you wanna get fancy, you could include some pictures on what is pointing to what, like a diagram or something

**Review by nizolnier — APPROVED** [review]
great job 😎

File: src/intro-to-c/pointers/using-pointers.md (hunk 3 above)
Comment (jerrettl) [reply src/intro-to-c/pointers/using-pointers.md:69]: I'll probably do it for version 2, I agree but I haven't decided how I want to do the diagrams yet

**Pull request merged**